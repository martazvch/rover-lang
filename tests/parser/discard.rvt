// Discard expression

code:
_ = if true do 5 else 10
_ = 56
_ = { 5 }

expect:
[Discard
    [If
        condition:
        [Bool literal true]
        then body:
        [Int literal 5]
        else body:
        [Int literal 10]
    ]
]
[Discard
    [Int literal 56]
]
[Discard
    [Block]
        [Int literal 5]
]

================

code:
_  1 + add()

error:
InvalidDiscard

================
