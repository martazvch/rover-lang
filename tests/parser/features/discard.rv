_ = if true do 5 else 10
_ = 56
_ = { 5 }

