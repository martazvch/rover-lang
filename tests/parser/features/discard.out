{
    "discard": {
        "if": {
            "condition": {
                "literal": "true"
            },
            "then": {
                "literal": "5"
            },
            "else": {
                "literal": "10"
            }
        }
    },
    "discard": {
        "literal": "56"
    },
    "discard": {
        "block": [
            "literal": "5"
        ]
    }
}
